packages/ckeditor5-engine/tests/model/utils/insertcontent.js

```diff
@@ -1299,6 +1299,1251 @@ describe( 'DataController utils', () => {
+					expect( expectedMarker.getRange().start.path ).to.deep.equal( [ 0, 3 ] );
+					expect( expectedMarker.getRange().end.path ).to.deep.equal( [ 0, 3 ] );
+				} );
+
+				it( 'should create collapsed marker after inserting imageBlock next to paragraph', () => {
+					setData( model, '<paragraph>foo</paragraph>[]' );
+
+					insertHelper( '<imageBlock></imageBlock>', {
+						'marker-a': { start: [ 0 ], end: [ 0 ] }
+					} );
+
+					const expectedMarker = model.markers.get( 'marker-a' );
+
+					// <paragraph>foo</paragraph>[<imageBlock></imageBlock>]
+					expect( getData( model ) ).to.equal( '<paragraph>foo</paragraph>[<imageBlock></imageBlock>]' );
+
+					// <paragraph>foo{}</paragraph><imageBlock></imageBlock>
+					expect( expectedMarker.getRange().start.path ).to.deep.equal( [ 0, 3 ] );
+					expect( expectedMarker.getRange().end.path ).to.deep.equal( [ 0, 3 ] );
+				} );
+
+				it( 'should create collapsed marker after inserting imageBlock before paragraph', () => {
+					setData( model, '[]<paragraph>foo</paragraph>' );
+
+					insertHelper( '<imageBlock></imageBlock>', {
+						'marker-a': { start: [ 0 ], end: [ 0 ] }
+					} );
+
+					const expectedMarker = model.markers.get( 'marker-a' );
+
+					// <paragraph></paragraph>[<imageBlock></imageBlock>]<paragraph>foo</paragraph>
+					expect( getData( model ) ).to.equal( '<paragraph></paragraph>[<imageBlock></imageBlock>]<paragraph>foo</paragraph>' );
+
+					// <paragraph>{}</paragraph><imageBlock></imageBlock><paragraph>foo</paragraph>
+					expect( expectedMarker.getRange().start.path ).to.deep.equal( [ 0, 0 ] );
+					expect( expectedMarker.getRange().end.path ).to.deep.equal( [ 0, 0 ] );
+				} );
+
+				it( 'should create multiple collapsed marker after inserting different elements in empty content', () => {
+					setData( model, '' );
+
+					insertHelper( '<imageInline></imageInline><imageBlock></imageBlock>', {
+						'marker-a': { start: [ 0 ], end: [ 0 ] },
+						'marker-b': { start: [ 1 ], end: [ 1 ] }
+					} );
+
+					const expectedMarkerA = model.markers.get( 'marker-a' );
+					const expectedMarkerB = model.markers.get( 'marker-b' );
+
+					// <paragraph><imageInline></imageInline></paragraph>[<imageBlock></imageBlock>]
+					expect( getData( model ) ).to.equal( '<paragraph><imageInline></imageInline></paragraph>[<imageBlock></imageBlock>]' );
+
+					// <paragraph>{}<imageInline></imageInline>()</paragraph><imageBlock></imageBlock>
+					expect( expectedMarkerA.getRange().start.path ).to.deep.equal( [ 0, 0 ] );
+					expect( expectedMarkerA.getRange().end.path ).to.deep.equal( [ 0, 0 ] );
+					expect( expectedMarkerB.getRange().start.path ).to.deep.equal( [ 0, 1 ] );
+					expect( expectedMarkerB.getRange().end.path ).to.deep.equal( [ 0, 1 ] );
+				} );
+			} );
+
+			it( 'should does not create marker when insertion is forbidden', () => {
+				setData( model, '<wrapper><limit><paragraph>[]</paragraph></limit></wrapper>', {
```

Comment:
Umm wait wait wait. Markers declaring should be in `insertHelper` not here :).

Reply:
What do you mean? Markers are sent in the second parameter